### PR TITLE
[Build] Add support for setuptools 69.0.0

### LINKED ISSUE #59: Remove `setup.py` and adjust `setuptools` version
## What

Installation via `setuptools` is currently broken. Remove `setup.py` and update `pyproject.toml` so we can still install with `setuptools` > `69.0.0`.

## Why

Installing from PyPI currently misses dependencies as those listed in `setup.py` are no longer considered.

## Notes

~According to https://github.com/pypa/setuptools/commit/3bd67c0e0f51da9823c074dc5645c3ab2284d16d we don't need to set `include-package-data`.  However, consumers might be stuck on an older `setuptools`, so we should still include it.~ [edit: if you include it, `setuptools` errors, telling you not to].

### PR DESCRIPTION
**Pull request by elliotcmorris**
Closes #59 

Setuptools 69.0.0 caused a break in our hybrid configurations where we used both `setup.py` and `pyproject.toml` to configure install-requirements. Move to just using `pyproject.toml`.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -36,6 +37,10 @@ classifiers = [
     "Topic :: Software Development :: Code Generators"
 ]
 
+[project.scripts]
+openassetio-traitgen = "openassetio_traitgen.__main__:main"
+
+
```
Comment (foundrytom) [inline pyproject.toml]: [Consider][minor]
```suggestion
```

**Review by foundrytom — APPROVED** [review]
Thanks for sorting that

File: pyproject.toml (hunk 1 above)
Comment (elliotcmorris) [reply pyproject.toml]: Caught it before you! [175b224](https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/pull/60/commits/175b224b656d99102b97bb3553f1b50bbee58a72)

**Review by feltech — APPROVED** [review]
LGTM, tested both generators locally with MediaCreation, all good.

**Pull request merged**